modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/BulkActions.js

```diff
@@ -96,6 +99,21 @@ function BulkActions({
 
 			setCurrentSidePanelActionPayload(sidePanelActionPayload);
 		}
+		else if (
+			onBulkActionItemClick &&
+			typeof onBulkActionItemClick === 'function'
+		) {
+			const selectedItems = items.filter((item) =>
+				selectedItemsValue.includes(item[selectedItemsKey])
+			);
+
+			onBulkActionItemClick({
+				action: actionDefinition,
+				selectedItemsData: selectedItems,
+				selectedItemsKey,
+				selectedItemsValue,
+			});
```

Comment:
`selectedItemsKey` is passed in all the way from main props, right? We don't need to add it here, we can reference it directly, like we do with `additionalProps` [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/SampleFDSPropsTransformer.js#L18). But we can, it's not a big deal.

I guess `selectedItemsValue` is handy, although it can be derived from the list of items and key.

It's interesting to have key as a prop. I would normally expect to construct request manually in frontend and be careful to match backend. With this, correct parameter can be set in a single place, in backend.

Maybe we can bundle selected items info into something like
```js
selectedItemsData: {
    items: selectedItems,
    key: selectedItemsKey,
    value: selectedItemsValue,
}
```

Reply:
> We don't need to add it here, we can reference it directly, like we do with additionalProps

Good point! However for your next comments I'm not sure how to proceed, I think it's better no to pass the key as it is available already...

Reply:
> I guess selectedItemsValue is handy, although it can be derived from the list of items and key.

I know but it's already done in the component so we can save that process and just pass the array

Reply:
> Maybe we can bundle selected items info into something like

I like that compact structure, pushing!